temperature/services/weather_api.py

```
@@ -0,0 +1,18 @@
+import httpx
+
+WEATHER_API_URL = "https://api.weatherapi.com/v1/current.json"
+
+class WeatherAPI:
+    def __init__(self, api_key: str) -> None:
+        self.api_key = api_key
+
+    async def get_temperature(self, city: str | None = None) -> float | None:
+        async with httpx.AsyncClient() as client:
+            params = {"key": self.api_key, "q": city}
+            try:
+                response = await client.get(WEATHER_API_URL, params=params)
+                response.raise_for_status()
+                data = response.json()
+                return data["current"]["temp_c"]
+            except Exception:
+                return None
```

The `except Exception:` block swallows all errors and returns `None`, which makes failures silent and makes debugging harder. Prefer catching specific exceptions (e.g. `httpx.HTTPError`, `ValueError`) and at minimum log the exception before returning `None`. That improves observability and aligns with the project requirement to handle errors gracefully.